bluesky/run_engine.py

```diff
@@ -1212,12 +1309,21 @@ def _stage(self, msg):
 
     @asyncio.coroutine
     def _unstage(self, msg):
+        """Instruct the RunEngine to unstage the object
+
+        Expected message object is:
+
+            Msg('unstage', object)
+        """
         _, obj, args, kwargs = msg
         # If an object has no 'unstage' method, assume there is nothing to do.
         if not hasattr(obj, 'unstage'):
             return
         result = obj.unstage()
```

Comment:
> Yeah, interesting thought, but I vote for #3. 

Alright. Works for me.

> The perf is negligible.

Is `_stage/_unstage` called at the beginning/end of the scan only?